vllm/distributed/parallel_state.py

```diff
@@ -446,15 +446,15 @@ def recv_object(self, src: int) -> Any:
 
         assert src < self.world_size, f"Invalid src rank ({src})"
 
-        assert src != self.rank, (
+        assert src != self.rank_in_group, (
             "Invalid source rank. Source rank is the same as the current rank."
         )
 
         size_tensor = torch.empty(1, dtype=torch.long, device="cpu")
 
         # Receive object size
         rank_size = torch.distributed.recv(size_tensor,
-                                           src=src,
+                                           src=self.ranks[src],
```

Comment:
my mind must be blurry now. I remember I comment this before in your previous pp helper pr :(

Reply:
Yes, you did comment this. We changed it in some of the locations but turns out we missed in a few other places. We miss this because we have a test for TP-only broadcast, PP-only send/recv, but no TP/PP combined test.

Reply:
I added the TP/PP combined comm ops test in this PR so we can test this is correct. This doubles the amount of tests for that though. Let me know if you still want to include that.